File: tvr-decoder/src/main/java/com/willcurrie/hex/ByteElement.java
package com.willcurrie.hex;

import org.apache.commons.lang.builder.EqualsBuilder;
import org.apache.commons.lang.builder.HashCodeBuilder;
import org.apache.commons.lang.builder.ToStringBuilder;
import org.apache.commons.lang.builder.ToStringStyle;

public class ByteElement implements HexDumpElement {

    private final String value;
    private final int byteOffset;

    public ByteElement(String value, int byteOffset) {
        this.value = value;
        this.byteOffset = byteOffset;
    }

    @Override
    public boolean isByte() {
        return true;
    }

    public int getByteOffset() {
        return byteOffset;
    }

    public String getValue() {
        return value;
    }

    @Override
    public int hashCode() {
        return HashCodeBuilder.reflectionHashCode(this);
    }

    @Override
    public boolean equals(Object obj) {
        return EqualsBuilder.reflectionEquals(this, obj);
    }

    @Override
    public String toString() {
        return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE);
    }
}

File: src/main/java/io/github/binaryfoo/hex/ByteElement.kt
package io.github.binaryfoo.hex

import org.apache.commons.lang.builder.EqualsBuilder
import org.apache.commons.lang.builder.HashCodeBuilder
import org.apache.commons.lang.builder.ToStringBuilder
import org.apache.commons.lang.builder.ToStringStyle

public data class ByteElement(val value: String, val byteOffset: Int) : HexDumpElement {

    override fun isByte(): Boolean {
        return true
    }

}
